llama_index/response_synthesizers/refine.py

```diff
@@ -25,11 +76,24 @@ def __init__(
         refine_template: Optional[RefinePrompt] = None,
         streaming: bool = False,
         verbose: bool = False,
+        structured_answer_filtering: bool = False,
+        program_factory: Optional[Callable[[Prompt], BasePydanticProgram]] = None,
```

Comment:
nit: it's not immediately obvious to me when and how to set `program_factory`

Reply:
Oh I see it's useful for testing.

Reply:
Its def useful for testing but I can also imagine a real use case for it. It opens up the possibility for non-OpenAI function calling users to provide their own `Program` implementations using their own custom prompts.